ingest-app/src/cmr/ingest/services/subscriptions_helper.clj

```diff
@@ -189,15 +189,47 @@
           (postal-core/send-message email-settings email-content)
           (info (str "Successfully processed subscription [" sub-id "].
                       Sent subscription email to [" email-address "]."))
+          (send-update-subscription-notification-time! context sub-id)
           (catch Exception e
             (error "Exception caught in email subscription: " sub-id "\n\n"
                    (.getMessage e) "\n\n" e)))))))
 
 (defn email-subscription-processing
   "Process email subscriptions and send email when found granules matching the collection and queries
   in the subscription and were created/updated during the last processing interval."
-  [context]
-  (let [subscriptions (->> (mdb/find-concepts context {:latest true} :subscription)
-                           (remove :deleted)
-                           (map #(select-keys % [:concept-id :extra-fields :metadata])))]
-    (send-subscription-emails context (process-subscriptions context subscriptions))))
+  ([context]
+   (email-subscription-processing context nil))
+  ([context revision-date-range]
+   (let [subscriptions (->> (mdb/find-concepts context {:latest true} :subscription)
+                            (remove :deleted)
+                            (map #(select-keys % [:concept-id :extra-fields :metadata])))]
+     (send-subscription-emails context (process-subscriptions context subscriptions revision-date-range)))))
+
+(defn- throw-error-if-one-sided-date-map
+  "Throws a service error if the provided map does not contain both keys :start-date and :end-date"
+  [date-map]
+  (when (not-every? date-map [:start-date :end-date])
+    (errors/throw-service-error :bad-request "One-sided date formats are not permitted. Please provide a start and end date.")))
+
+(defn- throw-error-if-start-date-after-end-date
+  "Throws a service error if the start date is after the end date"
+  [start-date end-date]
+  (when (t/after? start-date end-date)
+    (errors/throw-service-error :bad-request "The start date should occur before the end date.")))
+
+(defn- validate-revision-date-range
+  "Throws service errors on invalid revision date ranges"
+  [revision-date-range]
+  (let [revision-date-map (date-time-range-parser/parse-datetime-range revision-date-range)
+        _ (throw-error-if-one-sided-date-map revision-date-map)
+        start-date (:start-date revision-date-map)
+        end-date (:end-date revision-date-map)]
+    (throw-error-if-start-date-after-end-date start-date end-date)))
```

Comment:
nit: put all validation in this function rather than calling more functions. i.e.:
```
(let [{:keys [start-date end-date]} (date-time-range-parser/parse-datetime-range revision-date-range)]
  (when-not start-date
    (errors/throw-service-error :bad-request "Missing start date in revision-date-range."))
  (when-not end-date
    (errors/throw-service-error :bad-request "Missing end date in revision-date-range."))
  (when (or (t/equal? start-date end-date)
            (t/after? start-date end-date))
    (errors/throw-service-error :bad-request "The start date should occur before the end date.")))
```